### PR TITLE
Use SVG sprites for icons

### PR DESCRIPTION
**Pull request by SleeplessOne1917**

### CODE REVIEW

File: src/ui/components/common/icon.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      IconType::Search => "search",
+      IconType::Upvote => "upvote",
+      IconType::VerticalDots => "vertical-dots",
+    }
+  }
+}
+
+#[component]
+pub fn Icon(
+  #[prop(into)] icon: MaybeSignal<IconType>,
```
Comment (dessalines) [inline src/ui/components/common/icon.rs:43]: This doesn't need to be a signal I don't think. Doesn't need state / can be made at compile time.

**Review by dessalines — APPROVED** [review]
Feel free to merge whenev

File: src/ui/components/common/icon.rs (hunk 1 above)
Comment (SleeplessOne1917) [reply src/ui/components/common/icon.rs:43]: I could see a scenario where there is one `Icon` component that has its icon type change. `MaybeSignal` accepts both signal and non-signal attributes.

**Comment by SleeplessOne1917** [comment]
Is there a way to bypass CI for merging? Woodpecker always seems to act strange with this project.

**Pull request merged**